### DISCUSSION TITLE
how to get O1-preview or O1-mini working?

### DISCUSSION THREAD

**Discussion by JackPo** [huggingface/chat-ui#1540] [Q&A]
I get errors like:
400 Unsupported value: 'messages[0].role' does not support 'system' with this model.

**Comment by evalstate** [comment]
Hi JackPo - not sure if you have an answer for this yet.

ChatUI 0.9.4 has added a configuration option to disable the System Prompt - unfortunately this doesn't work for the o1 models because they use the Chat rather than Completion code path.

I have added rudimentary support to my fork here : https://github.com/barrensconsulting/llmindset-hug-ui if you want to try it.

![chatui-o1-support](https://github.com/user-attachments/assets/8a79ec55-ab03-4cbe-81d4-fe10dcabb607)

In that build, if you configure o1 similar to the below, then you will get an expandable display showing token usage including reasoning tokens (shown in the shot above). o1 doesn't support streaming, tools or images. I'll add to the discussion at https://github.com/huggingface/chat-ui/pull/1539 - but for the moment until OpenAI move the model from Beta I'll probably leave as-is with basic support on my fork. I don't know if @nsarrazin already has plans for this area?

```
 {   "name": "o1-mini",

       ... rest of model config ...

      "systemRoleSupported": false,
      "tokenInfo": { "contextWindow": 128000, "pricing": {"input": 3.0, "output": 12.00, "cached": 1.50 }},
      "tools": false,
      "endpoints":
      [{"type": "openai", "apiKey": "sk-YOUR-KEY-HERE", "baseURL": "https://api.openai.com/v1"}]
  }]  
```
